internal/instrument/instrument_test.go

```diff
@@ -877,3 +877,58 @@ func register() {
 		})
 	}
 }
+
+func TestEcho(t *testing.T) {
```

Comment:
Doesn't have to be done in this PR and it is not urgent, but there is an opportunity to refactor the tests and extracts the common code.
For instance, `TestGin` and `TestEcho` are very similar, we can imagine having a parent test suite `TestMiddleware` that contains the common code and calls Gin's and Echo's test cases.

Reply:
@ahmed-mez I'll do it in a separate PR for all the multiple low hanging refactors that we already discussed in this PR.